pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -0,0 +1,298 @@
+        # if we compare "y" then we are comparing the actual sales
+        if variable not in ["mu", "y"]:
+            raise ValueError(f"variable must be either 'mu' or 'y', not {variable}")
+
+        return (
+            self.idata.posterior_predictive["mu"]
+            - self.idata_counterfactual.posterior_predictive["mu"]
+        )
+
+    def plot_fit(self, variable="mu"):
+        """Plots the model fit (posterior predictive) of the background products."""
+
+        if variable not in ["mu", "y"]:
+            raise ValueError(f"variable must be either 'mu' or 'y', not {variable}")
+
+        fig, ax = plt.subplots()
+
+        # plot data
+        self.plot_data(self.data, ax)
+
+        # plot posterior predictive distribution of sales for each of the background products
+        x = self.data.index.values
+        background_products = list(self.idata.observed_data.background_product.data)
+        for i, background_product in enumerate(background_products):
+            az.plot_hdi(
+                x,
+                self.idata.posterior_predictive[variable]
+                .transpose(..., "time")
+                .sel(background_product=background_product),
+                fill_kwargs={
+                    "alpha": HDI_ALPHA,
+                    "color": f"C{i}",
+                    "label": "Posterior predictive (HDI)",
+                },
+                smooth=False,
+            )
+
+        # formatting
+        ax.legend()
+        ax.set(title="Model fit of sales of background products", ylabel="Sales")
+        return ax
+
+    def plot_counterfactual(self, variable="mu"):
+        """Plots the predicted sales of the background products under the counterfactual
+        scenario of never releasing the new product."""
+        fig, ax = plt.subplots()
+
+        if variable not in ["mu", "y"]:
+            raise ValueError(f"variable must be either 'mu' or 'y', not {variable}")
+
+        # plot data
+        self.plot_data(self.data, ax)
+
+        # plot posterior predictive distribution of sales for each of the background products
+        x = self.data.index.values
+        background_products = list(self.idata.observed_data.background_product.data)
+        for i, background_product in enumerate(background_products):
+            az.plot_hdi(
+                x,
+                self.idata_counterfactual.posterior_predictive[variable]
+                .transpose(..., "time")
+                .sel(background_product=background_product),
+                fill_kwargs={
+                    "alpha": HDI_ALPHA,
+                    "color": f"C{i}",
+                    "label": "Posterior predictive (HDI)",
+                },
+                smooth=False,
+            )
+
+        # formatting
+        ax.legend()
+        ax.set(
+            title="Model predictions under the counterfactual scenario", ylabel="Sales"
+        )
+        return ax
+
+    def plot_causal_impact(self, type="sales"):
```

Comment:
Will type ever be more then these two values? If not, breaking out into two methods might allow for easier discover

Also not fan of name type as it is a Python keyword

Reply:
Thanks. Have split up into two different plot methods